parser: use second last token for errors when unimplemented

otan opened the pull request:
Commonly when there is an unimplemented error, the error message
displayed is not as helpful as the `lastTok` is past the obvious point
of error, e.g.:

```
-- xml unsupported
root@127.0.0.1:26257/defaultdb> create table tbl (x xml);
invalid syntax: statement ignored: at or near ")": syntax error: unimplemented: this syntax
SQLSTATE: 0A000
DETAIL: source SQL:
create table tbl (x xml)
                       ^
HINT: You have attempted to use a feature that is not yet implemented.
See: https://go.crdb.dev/issue-v/43355/v22.1

-- comment on extension unsupported
root@127.0.0.1:26257/defaultdb> comment on extension "bob";
invalid syntax: statement ignored: at or near "bob": syntax error: unimplemented: this syntax
SQLSTATE: 0A000
DETAIL: source SQL:
comment on extension "bob"
                     ^
HINT: You have attempted to use a feature that is not yet implemented.

Please check the public issue tracker to check whether this problem is
already tracked. If you cannot find it there, please report the error
with details by creating a new issue.

If you would rather not post publicly, please contact us directly
using the support form.

We appreciate your feedback.
```

This change moves the cursor back one token to make it more clear where
the error is:
```
-- xml
root@127.0.0.1:26257/defaultdb> create table tbl (x xml);
invalid syntax: statement ignored: at or near "xml": syntax error: unimplemented: this syntax
SQLSTATE: 0A000
DETAIL: source SQL:
create table tbl (x xml)
                    ^
HINT: You have attempted to use a feature that is not yet implemented.
See: https://go.crdb.dev/issue-v/43355/v22.1
-- comment on extension
root@127.0.0.1:26257/defaultdb> comment on extension "bob";
invalid syntax: statement ignored: at or near "extension": syntax error: unimplemented: this syntax
SQLSTATE: 0A000
DETAIL: source SQL:
comment on extension "bob"
           ^
HINT: You have attempted to use a feature that is not yet implemented.

Please check the public issue tracker to check whether this problem is
already tracked. If you cannot find it there, please report the error
with details by creating a new issue.

If you would rather not post publicly, please contact us directly
using the support form.

We appreciate your feedback.
```

Release note (sql change): Previously, when parsing an unimplemented SQL
syntax, we pointed the error to after the point in the syntax in which
it may be useful, e.g. for `CREATE TABLE tbl (x xml)`, we point at the
`)` and it seems as if `)` is unimplemented. The error now points at the
token beforehand - in the example this would be `xml`.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/75264)
<!-- Reviewable:end -->

otan committed:
parser: use second last token for errors when unimplemented

Commonly when there is an unimplemented error, the error message
displayed is not as helpful as the `lastTok` is past the obvious point
of error, e.g.:

```
-- xml unsupported
root@127.0.0.1:26257/defaultdb> create table tbl (x xml);
invalid syntax: statement ignored: at or near ")": syntax error: unimplemented: this syntax
SQLSTATE: 0A000
DETAIL: source SQL:
create table tbl (x xml)
                       ^
HINT: You have attempted to use a feature that is not yet implemented.
See: https://go.crdb.dev/issue-v/43355/v22.1

-- comment on extension unsupported
root@127.0.0.1:26257/defaultdb> comment on extension "bob";
invalid syntax: statement ignored: at or near "bob": syntax error: unimplemented: this syntax
SQLSTATE: 0A000
DETAIL: source SQL:
comment on extension "bob"
                     ^
HINT: You have attempted to use a feature that is not yet implemented.

Please check the public issue tracker to check whether this problem is
already tracked. If you cannot find it there, please report the error
with details by creating a new issue.

If you would rather not post publicly, please contact us directly
using the support form.

We appreciate your feedback.
```

This change moves the cursor back one token to make it more clear where
the error is:
```
-- xml
root@127.0.0.1:26257/defaultdb> create table tbl (x xml);
invalid syntax: statement ignored: at or near "xml": syntax error: unimplemented: this syntax
SQLSTATE: 0A000
DETAIL: source SQL:
create table tbl (x xml)
                    ^
HINT: You have attempted to use a feature that is not yet implemented.
See: https://go.crdb.dev/issue-v/43355/v22.1
-- comment on extension
root@127.0.0.1:26257/defaultdb> comment on extension "bob";
invalid syntax: statement ignored: at or near "extension": syntax error: unimplemented: this syntax
SQLSTATE: 0A000
DETAIL: source SQL:
comment on extension "bob"
           ^
HINT: You have attempted to use a feature that is not yet implemented.

Please check the public issue tracker to check whether this problem is
already tracked. If you cannot find it there, please report the error
with details by creating a new issue.

If you would rather not post publicly, please contact us directly
using the support form.

We appreciate your feedback.
```

Release note (sql change): Previously, when parsing an unimplemented SQL
syntax, we pointed the error to after the point in the syntax in which
it may be useful, e.g. for `CREATE TABLE tbl (x xml)`, we point at the
`)` and it seems as if `)` is unimplemented. The error now points at the
token beforehand - in the example this would be `xml`.

otan commented:
hmm, wrong way around!
